rust-toolchain

```diff
@@ -1 +1 @@
-nightly-2023-08-19
+nightly-2024-02-07
```

Comment:
I bumped to the most recent nightly b/c the most recent version of `cargo-chef` (used to build the image) requires `rustc >= 1.74`. Let me know if this is a concern, if so I can find a `cargo-chef` version to pin to in our release image.

Reply:
tests are failing can you check that out. this often happens when bumping the version

Reply:
I pushed the version back to the nightly release the day after rust 1.74.1 was released on stable, and the repo builds on the x86_64 Github Actions runner. There's still a failing test `test_default_config`, but this fails on main as well, so out of scope for this PR